src/components/AdobeSignDialog.vue

```diff
@@ -0,0 +1,156 @@
+              Please go to your inbox and complete the signing process.
+            </h3>
+            <h3>Alternatively, you can click the link below to go directly to the signing page:</h3>
+            <h4>
+              <a :href="adobeUrl" target="blank"> Sign the form now </a>
+            </h4>
+          </div>
+        </div>
+      </div>
+      <div v-if="!hasSigningUrl && docCreated">
+        <div class="flex flex-row">
+          <div class="justify-content-center flex">
+            <img src="../assets/loadinglion.webp" style="width: 40vh; padding: 0px" />
+          </div>
+          <div class="flex flex-column justify-content-center p-5">
+            <h3>We are creating your agreement document.</h3>
+            <h4>Please wait a few seconds</h4>
+          </div>
+        </div>
+      </div>
+    </template>
+  </PvDialog>
+</template>
+
+<script setup>
+import { ref, onMounted } from 'vue';
+import { useToast } from 'primevue/usetoast';
+import { useAuthStore } from '@/store/auth';
+import { storeToRefs } from 'pinia';
+
+const authStore = useAuthStore();
+const { roarfirekit } = storeToRefs(authStore);
+const initialized = ref(false);
+const agreementId = ref(null);
+const docStatus = ref(null);
+const adobeUrl = ref(null);
+const hasSigningUrl = ref(false);
+let unsubscribe;
+
+const init = () => {
+  if (unsubscribe) unsubscribe();
+  initialized.value = true;
+};
+
+unsubscribe = authStore.$subscribe(async (mutation, state) => {
+  if (state.roarfirekit.restConfig) init();
+});
+
+onMounted(() => {
+  if (roarfirekit.value.restConfig) init();
+});
+
+const signerEmail = ref('');
+const docCreated = ref(false);
+const isVisible = ref(false);
+const isAdult = ref(false);
+const emit = defineEmits(['consent-selected']);
+
+const toast = useToast();
+
+const props = defineProps({
+  isAdobe: { type: Boolean, required: true, default: false },
+  isAdult: { type: Boolean, required: true, default: false },
+  parentEmail: { type: String, required: false },
+});
+
+if (props.parentEmail) {
+  signerEmail.value = props.parentEmail;
+  createConsent();
+}
+
+isVisible.value = props.isAdobe;
+
+async function createConsent() {
+  agreementId.value = null;
+  docCreated.value = true;
+  let docType = props.isAdult ? 'Consent' : 'Assent';
+
+  agreementId.value = await authStore.createAdobeSignAgreement(signerEmail.value, docType);
+
+  await new Promise((resolve) => setTimeout(resolve, 5000));
+
+  if (agreementId.value) {
+    adobeUrl.value = await authStore.getAdobeSignSigningUrl(agreementId.value, signerEmail.value);
+    hasSigningUrl.value = true;
+  }
+
+  let tries = 0;
+
+  while (tries < 60) {
+    docStatus.value = await authStore.getAdobeSignAgreementStatus(agreementId.value);
+    if (docStatus.value !== 'SIGNED') {
+      tries += 1;
+      await new Promise((resolve) => setTimeout(resolve, 5000));
+    } else {
+      tries = 60;
+    }
+  }
+
+  if (docStatus.value === 'SIGNED') {
+    props.isAdobe = false;
```

Comment:
```suggestion
```
I am not sure that we need this, and I don't think Vue recommends changing the value of a prop.